src/Features/Core/Portable/CodeRefactorings/ConvertNumericLiteral/AbstractConvertNumericLiteralCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,151 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Editing;
+
+namespace Microsoft.CodeAnalysis.CodeRefactorings.ConvertNumericLiteral
+{
+    internal abstract class AbstractConvertNumericLiteralCodeRefactoringProvider : CodeRefactoringProvider
+    {
+        public sealed override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+        {
+            var document = context.Document;
+            var cancellationToken = context.CancellationToken;
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+            var numericToken = root.FindToken(context.Span.Start);
+            if (numericToken.ContainsDiagnostics)
+            {
+                return;
+            }
+
+            if (!numericToken.GetLocation().SourceSpan.IntersectsWith(context.Span))
+            {
+                return;
+            }
+
+            var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
+            if (!syntaxFacts.IsNumericLiteralExpression(numericToken.Parent))
+            {
+                return;
+            }
+
+            var numericText = numericToken.ToString();
+
+            const string DecimalPoint = ".";
+            if (numericText.Contains(DecimalPoint))
+            {
+                return;
+            }
+
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var valueOpt = semanticModel.GetConstantValue(numericToken.Parent);
+            if (!valueOpt.HasValue)
+            {
+                return;
+            }
+
+            var value = IntegerUtilities.ToInt64(valueOpt.Value);
```

Comment:
maybe we want to validate that this value is actual an integral value of some sort?  Then that would immediately eliminate cases with dots or 'e's when it was actually a float?  That might be simpler. Up to you.